test/s3select_test.cpp

```diff
@@ -2603,3 +2680,51 @@ TEST(TestS3selectFunctions, nested_query_multirow_result)
   s3select_res = run_s3select(input_query, input_csv);
   EXPECT_EQ(s3select_res, expected_res);
 }
+
+TEST(TestS3selectFunctions, opserialization_expressions)
+{
+  std::string input;
+  size_t size = 10;
+  generate_rand_columns_csv(input, size);
+
+  char a[5] = {'@', '#', '$', '%'};
+  char b[4] = {'!', '^', '&', '*'};
+  char x = a[rand() % 4];
+  char y = b[rand() % 4];
+
+  const std::string input_query = "select * from s3object ;";
+
+  run_s3select_test_opserialization(input_query, input, &x, &y);
+
+  const std::string input_query_1 = "select int(_1) from s3object where  nullif(_1, _2) is not null;";
+
+  std::string s3select_result_1 = run_s3select_opserialization_quot(input_query_1,input, true);
+
+  const std::string input_query_2 = "select int(_1) from s3object where int(_1) != int(_2);";
+
+  std::string s3select_result_2 = run_s3select(input_query_2,input);
+
+  std::string s3select_result_2_final = string_to_quot(s3select_result_2);
+
+  ASSERT_EQ(s3select_result_1, s3select_result_2_final);
+
+  const std::string input_query_3 = "select int(_1) from s3object where int(_1) != int(_2);";
+
+  std::string s3select_result_3 = run_s3select_opserialization_quot(input_query_3,input);
+
+  ASSERT_NE(s3select_result_1, s3select_result_3);
```

Comment:
the use of **ASSERT_NE** is problematic since it could be non-equal for other reasons (as we already observed)

**run_s3select_test_opserialization** is using the output of the previous query result as input for another query (CSV definitions are different per run)

the `select * from ..`  test is covering the multiple rows/multiple columns
the `string_to_quot`  test is covering multiple rows\single columns 

it needs to cover the ASNEEDED vs ALWAYS